packages/primitives/popover/stories/utils/with-option-panel.component.ts

```diff
@@ -99,26 +107,36 @@ export class WithEventBaseComponent implements AfterContentInit {
         return timeFromPrev;
     };
 
-    ngAfterContentInit() {
-        this.popoverContentDirective().onOpen.subscribe(this.onOpen);
-        this.popoverContentDirective().onClosed.subscribe(this.onClose);
-        this.popoverContentDirective().onOverlayOutsideClick.subscribe(this.onOverlayOutsideClick);
-        this.popoverContentDirective().onOverlayEscapeKeyDown.subscribe(this.onOverlayEscapeKeyDown);
-
-        /**
-         * There should be only one container. If there is more, en error is thrown.
-         */
-        this.containers = Array.from(this.elementRef.nativeElement?.querySelectorAll('.container') ?? []);
-        if (this.containers.length > 1) {
-            if (isDevMode()) {
-                console.error('<story>.elementRef.nativeElement', this.elementRef.nativeElement);
-                console.error('<story>.containers', this.containers);
-                throw Error('each story should have only one container!');
-            }
-        }
-        this.paramsContainers = Array.from(this.elementRef.nativeElement?.querySelectorAll('.ParamsContainer') ?? []);
+    constructor() {
+        afterNextRender({
```

Comment:
[afterNextRender](https://angularexperts.io/blog/new-angular-coding-style)

Here is a suggestion to use this coding style:

```
@Component({ /* ... */ })
export class MyComponent {
  #logProductChanges = effect(() => { /* ... */
  });
  #afterNextRenderRef = afterRender(() => { /* ... */
  });
  #afterNextRender = afterNextRender(() => { /* ... */
  });
}

```

Reply:
@pimenovoleg yes, this is something I also like but on the other hand we don't use references anywhere in the code so they are kinda useless. I am more into following the KISS and YAGNI rules - if you don't use it, we don't need it. It is also a matter of preferences in this case IMHO :) one like this while others like that. My preference is:

- I assign the reference when I need to call it
- typically, I create a method wrapper for the stuff and call it within the constructor

What do you think about that approach @pimenovoleg ? Of course,  I can align the code if we have strict rules for such situations :)

Reply:
I just had thoughts :) I agree that there are questions about this approach. We will wait for the styles update from the Angular team.

Reply:
Ok, cool, I am merging :)